### PR TITLE
Files v1

### PR DESCRIPTION
**Pull request by atreidas1**
Add filemodel, fileresource, file authorization.

### CODE REVIEW

File: items/models.py [hunk 1]
```diff
@@ -24,10 +24,22 @@ class Item(models.Model):
...
     db_table = 'todo_items'
     ordering = ['is_completed', 'deadline']
 
   def __str__(self):
     return self.description
+
+
+class File(models.Model):
+  filename = models.CharField(max_length=150, default='')
+  path = models.CharField(max_length=150, default='')
```
Comment (NikitaSedyx) [inline items/models.py:38]: for me filename and path are required fields

File: items/resources.py [hunk 2]
```diff
@@ -137,23 +169,32 @@ def prepend_urls(self):
...
+          auth.login(request,user)
+          return self.create_response(request, Unpacking.unpack_user(user))
+        else:
+          return self.create_response(request, {}, HttpForbidden )
       else:
-        return HttpResponse(status=500)
+        return self.create_response(request, {}, HttpUnauthorized)
     else:
-      return HttpResponse(status=500)
+      return self.create_response(request,Unpacking.unpack_user(current_user))
```
Comment (NikitaSedyx) [inline items/resources.py:189]: use space here, plz
`request,Unpacking.unpack_user(current_user)`

File: items/resources.py [hunk 3]
```diff
@@ -137,23 +169,32 @@ def prepend_urls(self):
...
   def login(self, request, **kwargs):
-    if request.method == 'POST':
-      data = json.loads(request.body)
-      username = data['username']
-      password = data['password']
+    self.method_check(request, allowed=['post'])
+    current_user = request.user
+    if current_user.is_anonymous():
+      data = self.deserialize(request, request.body,
+      format=request.META.get('CONTENT_TYPE', 'application/json'))
```
Comment (NikitaSedyx) [inline items/resources.py:176]: it's unreadable for me

File: items/resources.py [hunk 4]
```diff
@@ -137,23 +169,32 @@ def prepend_urls(self):
...
+        return self.create_response(request, {}, HttpUnauthorized)
     else:
-      return HttpResponse(status=500)
+      return self.create_response(request,Unpacking.unpack_user(current_user))
 
   def logout(self, request, **kwargs):
-    auth.logout(request)
-    return HttpResponse(status=200)
+    self.method_check(request, allowed=['get'])
+    if request.user and request.user.is_authenticated():
```
Comment (NikitaSedyx) [inline items/resources.py:193]: for what it's if?
https://docs.djangoproject.com/en/1.4/topics/auth/#how-to-log-a-user-out
_Note that logout() doesn’t throw any errors if the user wasn’t logged in._

File: items/tests.py [hunk 5]
```diff
@@ -65,9 +64,52 @@ def test_get_not_owner(self):
...
+      authentication=self.setSession('user2', 123)))
+
+
+class LoginResourceTest(ResourceTestCase):
+
+  def setUp(self):
+    super(LoginResourceTest, self).setUp()
+    self.login_url = '/api/v1/auth/login/'
+    self.logout_url = '/api/v1/auth/logout/'
+    user = User.objects.create_user('user1', 'user1@mail.ru', 123)
```
Comment (NikitaSedyx) [inline items/tests.py:81]: just `User.objects.create_user('user1', 'user1@mail.ru', 123)` is normal here for me

File: items/tests.py [hunk 6]
```diff
@@ -65,9 +64,52 @@ def test_get_not_owner(self):
...
+    self.assertHttpMethodNotAllowed(self.api_client.delete(self.login_url))
+
+  def test_get_login_MethodNotAllowed(self):
+    self.assertHttpMethodNotAllowed(self.api_client.get(self.login_url))
+
+  def test_put_login_MethodNotAllowed(self):
+    self.assertHttpMethodNotAllowed(self.api_client.put(self.login_url))
+
+  def test_post_login_Unauthorized(self):
+    self.assertHttpUnauthorized(self.api_client.post(self.login_url,format='json',data={'username':'user2','password':'123'}))
```
Comment (NikitaSedyx) [inline items/tests.py:96]: spaces between arguments

File: items/tests.py [hunk 7]
```diff
@@ -65,9 +64,52 @@ def test_get_not_owner(self):
...
+
+  def test_post_login_succes(self):
+    self.assertHttpOK(self.api_client.post(self.login_url,format='json',data={'username':'user1','password':'123'}))
```
Comment (ghost) [inline items/tests.py:99]: yeah, something wrong with formatting, it should looks like 

```
self.assertHttpOK(self.api_client.post(
    self.login_url, format='json', data={'username': 'user1', 'password': '123'}))
```

**Comment by ghost** [comment]
Reviewed, please take into account comments

**Pull request closed without merging**